samcli/commands/_utils/options.py

```diff
@@ -160,6 +160,33 @@ def image_repositories_callback(ctx, param, provided_value):
     return image_repositories if image_repositories else None
 
 
+def s3_bucket_arn(ctx, param, provided_value):
+    """
+    Check is s3 bucket is ARN or bucket name and return bucket name.
+    :param ctx: Click Context
+    :param param: Param name
+    :param provided_value: Value provided by Click, after being processed by user.
+    :return: s3 bucket name.
+    """
+
+    if provided_value is not None and provided_value.lower().startswith("arn"):
```

Comment:
If my bucket is named `arnold` or just `arn` (which I believe are valid bucket names), we enter this `if` and treat it as an ARN and probably fail with an error message.  I would suggest changing to `startswith("arn:")` since `:` is not valid for bucket names.

Side note: you could consider the [ArnParser](https://github.com/boto/botocore/blob/5093dccfecb2fa002f45fb18ab88eb076c5945f9/botocore/utils.py#L1467) class in botocore for more readability.